eax/eax.go

```diff
@@ -50,19 +49,18 @@ func NewEAX(key []byte) (cipher.AEAD, error) {
 // Only to be used for compatibility with existing cryptosystems with
 // non-standard parameters. For all other cases, prefer NewEAX.
 func NewEAXWithNonceAndTagSize(
-	key []byte, nonceSize, tagSize int) (cipher.AEAD, error) {
+	block cipher.Block, nonceSize, tagSize int) (cipher.AEAD, error) {
+	if block.BlockSize() != 16 {
+		return nil, eaxError("Block cipher must have 128-bit blocks")
+	}
```

Comment:
The eax paper doesn't require even this, do we assume it somewhere?

Reply:
Yeah, I think since it's a restriction of the OpenPGP spec whereas this aims to be a more general implementation of EAX, it could be an idea to do so.